test/Microsoft.ML.Tests/TrainerEstimators/TreeEnsembleFeaturizerTest.cs

```diff
@@ -108,5 +115,633 @@ public void TreeEnsembleFeaturizerOutputSchemaTest()
+
+            Assert.True(metrics.Accuracy > 0.97);
+            Assert.True(metrics.LogLoss < 0.07);
+            Assert.True(metrics.AreaUnderPrecisionRecallCurve > 0.98);
+        }
+
+        [Fact]
+        public void TestFastTreeRegressionFeaturizationInPipeline()
+        {
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(dataPointCount).ToList();
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var trainerOptions = new FastTreeRegressionTrainer.Options
+            {
+                    NumberOfThreads = 1,
+                    NumberOfTrees = 10,
+                    NumberOfLeaves = 4,
+                    MinimumExampleCountPerLeaf = 10,
+                    FeatureColumnName = "Features",
+                    LabelColumnName = "Label"
+            };
+
+            var options = new FastTreeRegressionFeaturizationEstimator.Options()
+            {
+                InputColumnName = "Features",
+                TreesColumnName = "Trees",
+                LeavesColumnName = "Leaves",
+                PathsColumnName = "Paths",
+                TrainerOptions = trainerOptions
+            };
+
+            var pipeline = ML.Transforms.FeaturizeByFastTreeRegression(options).
+                Append(ML.Transforms.Concatenate("CombinedFeatures", "Features", "Trees", "Leaves", "Paths")).
+                Append(ML.Regression.Trainers.Sdca("Label", "CombinedFeatures"));
+            var model = pipeline.Fit(dataView);
+            var prediction = model.Transform(dataView);
+            var metrics = ML.Regression.Evaluate(prediction);
+
+            Assert.True(metrics.MeanAbsoluteError < 0.2);
+            Assert.True(metrics.MeanSquaredError < 0.05);
+        }
+
+        [Fact]
+        public void TestFastForestRegressionFeaturizationInPipeline()
+        {
+            int dataPointCount = 200;
+            var data = SamplesUtils.DatasetUtils.GenerateFloatLabelFloatFeatureVectorSamples(dataPointCount).ToList();
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var trainerOptions = new FastForestRegressionTrainer.Options
+            {
+                    NumberOfThreads = 1,
+                    NumberOfTrees = 10,
+                    NumberOfLeaves = 4,
+                    MinimumExampleCountPerLeaf = 10,
+                    FeatureColumnName = "Features",
+                    LabelColumnName = "Label"
+            };
+
+            var options = new FastForestRegressionFeaturizationEstimator.Options()
+            {
+                InputColumnName = "Features",
+                TreesColumnName = "Trees",
+                LeavesColumnName = "Leaves",
+                PathsColumnName = "Paths",
+                TrainerOptions = trainerOptions
+            };
+
+            var pipeline = ML.Transforms.FeaturizeByFastForestRegression(options).
```

Comment:
Can you add a test of `ML.Transforms.FeaturizeByFastForestRegression()` using FastForest's [`ShuffleLabels`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.fasttree.fastforestregressiontrainer.options?view=ml-dotnet) option? I believe this is the only way to use TreeFeat w/ multi-class classification.

Reply:
Let's make it in another PR. This PR has been too large..

Reply:
Seems the UnitTests should be in the main PR.

Reply:
Thanks for adding.
#Resolved